### PR TITLE
Adds support for https repositories along with a script to generate certificates

### PR DESCRIPTION
**Pull request by kkaarreell**
There are 4 commits:
1st adds a bash script to generate certificates, it is slightly modified library written by Huber Kario, available at GitHub https://github.com/redhat-qe-security/certgen
2nd updates Dockerfile to build set of testing certificates
3rd configures httpd to serve files over https
4th updates repo_steps.py and repo_utils.py with support for https repositories

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -39,6 +40,23 @@ RUN set -x && \
...
+    /usr/local/bin/x509certgen x509KeyGen server2 && \
+    /usr/local/bin/x509certgen x509KeyGen client2 && \
+    /usr/local/bin/x509certgen x509SelfSign ca && \
+    /usr/local/bin/x509certgen x509SelfSign ca2 && \
+    /usr/local/bin/x509certgen x509CertSign --CA ca server && \
+    /usr/local/bin/x509certgen x509CertSign --CA ca client && \
+    /usr/local/bin/x509certgen x509CertSign --CA ca2 server2 && \
+    /usr/local/bin/x509certgen x509CertSign --CA ca2 client2 && \
+    # configure httpd
+    sed -i "s/#ServerName .*/ServerName ${HOSTNAME}:80/" /etc/httpd/conf/httpd.conf && \
```
Comment (ignatenkobrain) [inline Dockerfile]: hmmm, I miss where we define this variable... probably you need to use single quotes to not expand variables?

File: dnf-docker-test/features/steps/repo_steps.py [hunk 2]
```diff
@@ -244,6 +246,10 @@ def given_repository_with_packages(ctx, rtype, repository):
     ctx.table.add_row(["",         "enabled",  "False"])
     ctx.table.add_row(["",         "gpgcheck", "False"])
     ctx.table.add_row(["",         "baseurl",  "{!s}://{!s}".format(rtype, repopath)])
+    if rtype == 'https':
+        ctx.table.add_row(["",         "sslcacert",  "/etc/pki/tls/certs/testcerts/ca/cert.pem"])
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/steps/repo_steps.py]: don't bother with additional alignments ;)

I just wanted all parameters to be on the same level.

File: Dockerfile (hunk 1 above)
Comment (kkaarreell) [inline Dockerfile]: HOSTNAME variable is exported by bash itself... This change helps to get rid off the message 'httpd: Could not reliably determine the server's fully qualified domain name'

File: dnf-docker-test/features/steps/repo_steps.py (hunk 2 above)
Comment (kkaarreell) [inline dnf-docker-test/features/steps/repo_steps.py]: will fix

File: Dockerfile (hunk 1 above)
Comment (ignatenkobrain) [inline Dockerfile]: ah, right.

**Pull request merged**